docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,351 @@
+| AWI          | Azure Workload Identity |
+| OIDC         | OpenID Connect          |
+| JWKS         | JSON Web Key Sets       |
+
+## <a name='Summary'></a>Summary
+
+Workloads deployed in Kubernetes cluster may require Azure Active Directory application credential or managed identities to access azure protected resource e.g. Azure Key Vault, Virtual Machines etc. Azure AD Pod Identity helps access azure resources without the need of a secret management via Azure Managed Identities. Azure AD Pod Identity is now deprecated and Azure AD Workload Identity is the next iteration of the former. This design proposal aims to define the way for AWI integration into capz for self managed clusters with keeping in mind other factor e.g. User Experience, Multi Tenancy, Backwards Compatibility etc.
+
+For more details about Azure AD Pod Identity please visit this [link](#https://github.com/Azure/aad-pod-identity)  
+
+## <a name='Motivation'></a>Motivation
+
+AWI gives the capability to federate the credentials with external identity providers in a Kubernetes native way. This approach overcomes several limittations of Azure AD Identity as mentioned below.
+- Removes scale and performance issues that existed for identity assignment.
+- Supports K8s clusters hosted in any cloud or on premise.
+- Supports both Linux and Windows workloads and removes the need for CRD and pods that intercept IMDS traffic.
+
+To learn more about AWI please visit this link https://azure.github.io/azure-workload-identity/docs/introduction.html
+
+### <a name='Goals'></a>Goals
+
+- Integrate Azure Workload Identity into capz.
+
+### <a name='Non-GoalsFutureWork'></a>Non-Goals / Future Work
+
+- Migration to `azidentity` from `adal` module when using `go-autorest` library in capz as `adal` will be deprecated. 
+
+## <a name='Proposal'></a>Proposal
+
+In this model, Kubernetes cluster itself becomes the token issuer issuing tokens to Kubernetes service accounts. These service accounts can be configured to be trusted on Azure AD application or user assigned identity. Workloads/pods can use this service account token which is projected to it's volume and can exchange for an Azure AD token.
+
+The first step for creating K8s cluster using capz is creating a bootstrap cluster which is a Kind cluster and then deploying capi, capz and other helper components. On a high level the workflow looks the following to be able to use AWI and create a management cluster.
+
+**Bootstrap Cluster**
+
+- The operator/admin should be able generate signing key pair or BYO key pair. 
+
+- The bootstrap cluster i.e the Kind K8s cluster should be configured with appropriate flags on kube-apiserver and kube-controller-manager and the key pairs should be mounted on the container path for control plane node. 
+
+- The operator/admin should set up OIDC issuer URL which means having access to azure blob storage where the following two documents will be stored in the blob storage container.
```

Comment:
This assumes that storage account is already created, right? Also, along with container.

Reply:
Yes.